### PR TITLE
Disable chunk trimming in Receivers

### PR DESCRIPTION
**Pull request by fpetkovski**
This change is motivated by the following Slack discussion: https://cloud-native.slack.com/archives/C02KR205UMU/p1728575718863519

When trimming is not disabled, receivers end up recoding all chunks
in order to drop samples that are outside of the range.
This ends up being very expensive and causes ingestion problems during high
query load.

Trimming will be triggered in the following case:
```
|-------------| chunk
   |-------|        requested range
```

This commit disables trimming which should reduce CPU usage in receivers.
The engine already skips samples outside of the query range, so correctness will not be affected.

* [ ] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

### CODE REVIEW

File: pkg/store/tsdb.go [hunk 1]
```diff
@@ -272,9 +272,10 @@ func (s *TSDBStore) Series(r *storepb.SeriesRequest, seriesSrv storepb.Store_Ser
 	}
 
 	hints := &storage.SelectHints{
-		Start: r.MinTime,
-		End:   r.MaxTime,
-		Limit: int(r.Limit),
+		Start:           r.MinTime,
+		End:             r.MaxTime,
+		Limit:           int(r.Limit),
+		DisableTrimming: true,
```
Comment (pedro-stanaka) [inline pkg/store/tsdb.go:278]: Couldn't this produce wrong results if someone is still using the Prometheus engine?

Comment (fpetkovski) [inline pkg/store/tsdb.go:278]: Actually both engines will skip samples outside of the range, so we should be safe here.

**Review by GiedriusS — APPROVED** [review]

**Review by pedro-stanaka — APPROVED** [review]

**Comment by fpetkovski** [comment]
Forgot to sign DCO and had to force-push.

**Review by pedro-stanaka — APPROVED** [review]

**Review by GiedriusS — APPROVED** [review]

**Comment by pedro-stanaka** [comment]
Do we need changelog for this one?

**Review by bwplotka — APPROVED** [review]
Let's double check we are explicitly stating that in `Series` gRPC API too (that time matcher is "best-effort" to some extend).

**Comment by fpetkovski** [comment]
I've extended the godoc on the Series RPC to mention that chunks don't need to be within the exact range. I think this is good enough for now and we don't need a changelog entry because the change is not user facing.

**Comment by fpetkovski** [comment]
The engine will discard samples outside of the query range, so results are not affected in any way.

**Comment by fpetkovski** [comment]
@MichaHoffmann pointed me to the acceptance tests we run from the Prometheus repo itself: https://github.com/thanos-io/thanos/blob/main/pkg/query/query_test.go#L53.

So I think we should be good. Also store-gw has the same behavior.

**Comment by yeya24** [comment]
This option seems to have an edge case and it is captured by Cortex Query Fuzz test. See https://github.com/cortexproject/cortex/actions/runs/11621920479/job/32367371250?pr=6298

```
timestamp({__name__="test_series_3"} @ end() offset 3m21s)
```

This function returns different results after chunk trimming is disabled because now samples outside of the requested time range got returned and promql engine uses timestamp of a different timestamp now.

I am not sure if it is a bug or an expected behavior. I think we should put it under a FF for a behavior change like that.

**Comment by fpetkovski** [comment]
Thanks for reporting this. I will take a look on Monday why this happens. Honestly another feature flag for this would just add to the complexity that we have, so I would like to see if we can solve it without adding another knob to configure.

**Comment by yeya24** [comment]
If it is a bug of promql engine let's fix it.
If it is an intended behavior and we don't want to put this under a FF, let's just revert it.

**Comment by fpetkovski** [comment]
@yeya24 one more thing, is this failure happening with the prometheus or thanos engine?

**Comment by yeya24** [comment]
With Prometheus engine. I haven't ran the same test against thanos engine
